frontend/src/main/web/lib/components/glossary/DataTable.jsx

```diff
@@ -0,0 +1,438 @@
+    var footerHeight = footer ? footer.clientHeight : 91;
+    var top = _.isUndefined(fixedTop) ? React.findDOMNode(this).offsetTop: fixedTop;
+    var newHeight = window.innerHeight - footerHeight - top;
+
+    //minimum height 250px
+    newHeight = newHeight < 250 ? 250 : newHeight;
+    return newHeight;
+  },
+
+  _getWidth: function () {
+    return window.innerWidth - 48;
+  },
+
+  _handleResize: function(e) {
+    this.setState({tbl_height: this._getHeight(), tbl_width: this._getWidth()});
+  },
+
+  componentDidMount: function() {
+    window.addEventListener('resize', this._handleResize);
+  },
+
+  componentWillUnmount: function() {
+    window.removeEventListener('resize', this._handleResize);
+  },
+
+  _generateTermInfo: function(term) {
+    var title = "";
+    if(!_.isUndefined(term) && !_.isNull(term)) {
+      if (!StringUtils.isEmptyOrNull(term.lastModifiedBy)
+        || !StringUtils.isEmptyOrNull(term.lastModifiedDate)) {
+        title = "Last updated ";
+        if (!StringUtils.isEmptyOrNull(term.lastModifiedBy)) {
+          title += "by: " + term.lastModifiedBy;
+        }
+        if (!StringUtils.isEmptyOrNull(term.lastModifiedDate)) {
+          title += " " + term.lastModifiedDate;
+        }
+      }
+    }
+    if(StringUtils.isEmptyOrNull(title)) {
+      title = "No information available";
+    }
+    return title;
+  },
+
+  _generateKey: function (colIndex, rowIndex, resId) {
+    var key = colIndex + ":" + rowIndex + ":" + resId;
+    if(this.props.selectedTransLocale) {
+      key += ":" + this.props.selectedTransLocale;
+    }
+    return key;
+  },
+
+  _getSort: function (key) {
+    if(_.isUndefined(this.props.sort[key])) {
+      return null;
+    } else if(this.props.sort[key] === true) {
+      return "ascending";
+    } else {
+      return "descending";
+    }
+  },
+
+  _renderSourceHeader: function (label) {
+    var key = this.ENTRY.SRC.sort_field, asc = this._getSort(key);
+    return this._renderHeader(label, key, asc, true);
+  },
+
+  _renderTransHeader: function (label) {
+    var key = this.ENTRY.TRANS.sort_field, asc = null;
+    return this._renderHeader(label, key, asc, false);
+  },
+
+  _renderPosHeader: function (label) {
+    var key = this.ENTRY.POS.sort_field, asc = this._getSort(key);
+    return this._renderHeader(label, key, asc, true);
+  },
+
+  _renderDescHeader: function (label) {
+    var key = this.ENTRY.DESC.sort_field, asc = this._getSort(key);
+    return this._renderHeader(label, key, asc, true);
+  },
+
+  _renderTransCountHeader: function (label) {
+    var key = this.ENTRY.TRANS_COUNT.sort_field, asc = this._getSort(key);
+    return this._renderHeader(label, key, asc, true);
+  },
+
+  _renderHeader: function (label, key, asc, allowSort) {
+    return (<ColumnHeader value={label}
+      field={key}
+      key={key}
+      allowSort={allowSort}
+      sort={asc}
+      onClickCallback={this._onHeaderClick}/>);
```

Comment:
Could increase readability by accepting an object instead of 4 different parameters in this function. Could then use spread attributes to more concisely apply them to the `ColumnHeader`:

```
renderHeader: (opts) => {
  return <ColumnHeader {...opts}
               field={opts.key}
               onClickCallback={this._onHeaderClick}/>
}
```

makes it easier to read at the call site, since you can see what each thing is without having to look at the implementation:

```
this.renderHeader({ value: label, key: key, sort: asc, allowSort: true })
```